src/classes/MyHandler/offer/accepted/updateListings.ts

```diff
@@ -44,6 +44,9 @@ export default function updateListings(
         const isDisabledHV = highValue.isDisableSKU.includes(sku);
         const isAdmin = bot.isAdmin(offer.partner);
         const isNotSkinsOrWarPaint = SKU.fromString(sku).wear === null;
+        // if item is unusual and autoAddInvalidUnusual is set to true then we allow addInvalidUnusual.
+        // If the item is not an unusual sku, we "allow" still (no-op)
+        const addInvalidUnusual = SKU.fromString(sku).quality === 5 ? opt.pricelist.autoAddInvalidUnusual : true;
```

Comment:
This is mainly for the Generic Unusual feature right?
So I guess it's much better to do it like this (together with some small refactorization):

```ts
const item = SKU.fromString(sku);
const isNotSkinsOrWarPaint = item.wear === null;
const addInvalidUnusual = item.quality === 5 && item.effect === null ? opt.pricelist.autoAddInvalidUnusual : true;
```

Reply:
oh, wait. I don't think `item.effect === null` is correct.

Reply:
Yeah, scratch that. My bad.

Reply:
yeah I was about to say. I think in this case the SKU detected would be the SKU of the actual item at this point, so it wuld have an effect

Reply:
Yep correct. It's not like when receiving the painted item (with `normalize.painted.their` set to `false`)

Reply:
this is somewhat a strange statement but it really should be "true" if either:

the item is unsual and we autoAddInvalidUnusual or,
if the item isn't an unusual.